components/brave_sync/brave_profile_sync_service_impl.cc

```diff
@@ -672,7 +683,7 @@ int BraveProfileSyncServiceImpl::GetDisableReasons() const {
   // legacy sync only support bookmark sync so we have to wait for migration
   // complete before enable sync engine
   if (IsBraveSyncEnabled() &&
-      brave_sync_prefs_->GetMigratedBookmarksVersion() >= 2)
+      brave_sync_prefs_->GetMigratedBookmarksVersion() >= 3)
```

Comment:
We don't need to wait for version 3 migration, the reason we need to wait for version 2 is we need to set order for permanent node before sync engine is running